Gems/ScriptCanvas/Code/Editor/Framework/Configuration.cpp

```diff
@@ -0,0 +1,339 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzCore/Asset/AssetManagerBus.h>
+#include <AzCore/Component/ComponentApplicationBus.h>
+#include <AzCore/Serialization/EditContext.h>
+#include <AzCore/Serialization/IdUtils.h>
+#include <AzCore/Serialization/Json/RegistrationContext.h>
+#include <AzCore/Serialization/Utils.h>
+#include <AzFramework/Asset/AssetSystemBus.h>
+#include <AzToolsFramework/API/EditorAssetSystemAPI.h>
+#include <AzToolsFramework/API/ToolsApplicationAPI.h>
+#include <LyViewPaneNames.h>
+#include <ScriptCanvas/Assets/ScriptCanvasFileHandling.h>
+#include <Editor/Framework/Configuration.h>
+#include <ScriptCanvas/Components/EditorUtils.h>
+#include <ScriptCanvas/Bus/RequestBus.h>
+
+namespace ScriptCanvasEditor
+{
+    class OnScopeEnd
+    {
+    public:
+        using ScopeEndFunctor = std::function<void()>;
+
+    private:
+        ScopeEndFunctor m_functor;
+
+    public:
+        OnScopeEnd(ScopeEndFunctor&& functor)
+            : m_functor(AZStd::move(functor))
+        {}
+
+        OnScopeEnd(const ScopeEndFunctor& functor)
+            : m_functor(functor)
+        {}
+
+        ~OnScopeEnd()
+        {
+            m_functor();
+        }
+    };
+
+    Configuration::Configuration()
+        : Configuration(SourceHandle())
+    {
+    }
+
+    Configuration::Configuration(const SourceHandle& sourceHandle)
```

Comment:
The Configuration class takes on the extra functionality that used to be provided by the ScriptCanvasEditorComponent, and now exposes it for whatever classes need it. It allows the user to select a source graph, and configure its properties if possible. It communicates with the BuilderDataSystem to provide the property values, and status of the source script the user has selected (removed, failed, compiled, etc). It is used by the ScriptCanvasEditorComponent and the InterpreterWidget.